README.md

```diff
@@ -1,20 +1,27 @@
 
-NOTE: This project is in the initial stages of development.
+An in-depth description of AirScript is available in the full AirScript [documentation](https://0xpolygonmiden.github.io/air-script/).
+
+**WARNING**: This project is in an alpha stage. It has not been audited and may contain bugs and security flaws. This implementation is NOT ready for production use.
 
 ## Overview
 
-AirScript is a domain specific language for writing AIR constraints for the STARK proving system. The primary goals of AirScript is to enable writing and auditing AIR constraints without the need to learn a specific programming language (e.g., Rust). The secondary goal is to perform automated optimizations of constraints and to output constraint evaluator code to multiple backends (e.g., Rust, Miden assembly, Solidity etc.).
+AirScript is a domain-specific language for writing AIR constraints for the STARK proving system. The primary goal of AirScript is to enable writing and auditing AIR constraints without the need to learn a specific programming language (e.g., Rust). The secondary goal is to perform automated optimizations of constraints and to output constraint evaluator code in multiple target languages (e.g., Rust, Miden assembly, Solidity etc.).
 
 ## Project Structure
 
-The project is organized into several crates like so:
-| Crate                  | Description |
+The project is organized into several crates as follows:
+| Crate | Description |
 | ---------------------- | ----------- |
-| [Parser](parser) | Contains parser for the AirScript. The parser is used to parse the constraints written in AirScript into an AST. |
+| [Parser](parser) | Contains the parser for AirScript. The parser is used to parse the constraints written in AirScript into an AST. |
+| [IR](ir) | Contains the IR for AirScript, `AirIR`. `AirIR` is initialized with an AirScript AST, which it converts to an internal representation that can be optimized and used to generate code in multiple target languages. |
+| [Winterfell code generator](codegen-winter) | Contains a code generator targeting the [Winterfell prover](https://github.com/novifinancial/winterfell) Rust library. The Winterfell code generator converts a provided AirScript `AirIR` into Rust code that represents the AIR as a new custom struct that implements Winterfell's `Air` trait. |
 
 ## References
-1. [Lalrpop](https://github.com/lalrpop/lalrpop/): Rust parser generator framework.
-2. [Logos](https://github.com/maciejhirsz/logos/): Library to help in creating a lexer.
\ No newline at end of file
+
+1. [Logos](https://github.com/maciejhirsz/logos/): Library for generating fast lexers in Rust.
+1. [LALRPOP](https://github.com/lalrpop/lalrpop/): LR(1) Rust parser generator framework.
+1. [Codegen](https://github.com/carllerche/codegen): Library for generating Rust code.
+1. [mdBook](https://github.com/rust-lang/mdBook): Utility for creating online documentation books.
```

Comment:
nit: the numbering here should probably be 1, 2, 3, 4.

Also, I'd add a "License" section at the end of this README to similar to how we have it in the Miden repo.

Reply:
I usually do my markdown numbers like this because it renders as an ordered list (1, 2, 3, 4) and it's easier to update this way, but I don't mind changing it.

Reply:
I'm fine either way.